js/tests/Test_jsoo.ml

```diff
@@ -68,6 +68,18 @@ let _ =
        method setJsonnetParser = Semgrep_js_shared.setJsonnetParser
 
        method run filter =
+         (* This should place us at the root of the semgrep repository, which
+            is important for some assumptions later on during testing.
+         *)
+         let repo_root =
+           match Git_wrapper.get_project_root () with
+           | Some path -> path
+           | None ->
+               failwith
+                 "You must run the test program from within the semgrep repo \
+                  and not one of its submodules."
+         in
+         Unix.chdir (Fpath.to_string repo_root);
```

Comment:
You're not allowed to use chdir alone because it modifies the global state (and even if you're running the test in its own process, it's not obvious to the reader that it's safe). Use `Testutil_files.with_chdir` or equivalent.